lib/ansible/plugins/cache/__init__.py

```diff
@@ -29,10 +30,27 @@
 from ansible.module_utils.common._collections_compat import MutableMapping
 from ansible.plugins.loader import cache_loader
 from ansible.utils.display import Display
+from ansible.vars.fact_cache import FactCache as RealFactCache
 
 display = Display()
 
 
+class FactCache(RealFactCache):
```

Comment:
I had plans to remove this altogether from vars manager, not sure if it is worth moving. We should deprecate it either way.

Reply:
Yeah... it seems like just a shim.  On the other hand, it gives us dict-semantics which is very nice for using the fact cache.  It feels like the cache plugins themselves should have implemented the dict-API but since they didn't this was built as a middle layer to make it easier to consume those.

Reply:
originally it compensated for cache plugins not having dict semantics, but since then we updated the base classes to follow those. So this shim should not be needed anymore, though we don't inherit from mutablemapping in BaseCacheModule so we might still have work to do on that front.

Reply:
cache plugins don't yet have dict semantics.  They provide all of the building blocks but they are using the wrong method names and lack small bits of code which would make them usable as dicts.

Since this is not used by cache plugins it makes sense to move this now.  It also lets us get a headstart on deprecating it from the public plugin API location which is nice.  I could change the wording, though, to tell people to port to using the cache plugin directly via ```loader.cache_loader()``` if you want, though.